WalletWasabi/Wallets/Wallet.cs

```diff
@@ -410,6 +410,11 @@ private async void IndexDownloader_NewFilterAsync(object? sender, FilterModel fi
 				if (KeyManager.GetBestHeight() < filterModel.Header.Height)
 				{
 					await ProcessFilterModelAsync(filterModel, CancellationToken.None).ConfigureAwait(false);
+					if (KeyManager.GetBestHeight() < filterModel.Header.Height)
+					{
+						var filterHeight = new Height(filterModel.Header.Height);
+						KeyManager.SetBestHeights(filterHeight, filterHeight);
+					}
```

Comment:
This code is duplicated and it the one that should have its own method. I would do something like this:

```c#
private void SetWalletHeights(Height newHeight)
{
	if (KeyManager.GetBestHeight() < newHeight)
	{
		KeyManager.SetBestHeights(newHeight, newHeight);
	}
}
```

So here you can call:

```c#
SetWalletHeights(filterModel.Header.Height);
```

The same below.